lib/hidapi/udev.py

```diff
@@ -89,21 +91,24 @@ def _match(action, device, filter):
     product_id = filter.get('product_id')
     interface_number = filter.get('usb_interface')
     hid_driver = filter.get('hid_driver')
+    isDevice = filter.get('isDevice')
+    bus_id = filter.get('bus_id')
 
-    usb_device = device.find_parent('usb', 'usb_device')
-    # print ("* parent", action, device, "usb:", usb_device)
-    if not usb_device:
+    hid_device = device.find_parent('hid')
+    if not hid_device:
         return
+    hid_id = hid_device.get('HID_ID')
+    bid, vid, pid = hid_id.split(':') if hid_id else (None, None, None)
 
-    vid = usb_device.get('ID_VENDOR_ID')
-    pid = usb_device.get('ID_MODEL_ID')
     if vid is None or pid is None:
         return  # there are reports that sometimes the usb_device isn't set up right so be defensive
     if not ((vendor_id is None or vendor_id == int(vid, 16)) and (product_id is None or product_id == int(pid, 16))):
         return
+    if not (bus_id is None or bus_id == int(bid, 16)):
+        return
 
     if action == 'add':
-        hid_device = device.find_parent('hid')
+        ##        hid_device = device.find_parent('hid')
         if not hid_device:
```

Comment:
This is same check as line 98?

Reply:
Yes.  I didn't remove the check when I changed the logic.

Reply:
Fixed now.